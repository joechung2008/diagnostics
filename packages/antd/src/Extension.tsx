import { Typography } from "antd";
import Configuration from "./Configuration";
import StageDefinition from "./StageDefinition";

const Extension: React.FC<ExtensionProps> = ({
  config,
  extensionName,
  stageDefinition,
}) => {
  return (
    <div className="extension-root grow">
      <Typography.Title level={1}>{extensionName}</Typography.Title>
      {config && <Configuration config={config} />}
      {stageDefinition && <StageDefinition stageDefinition={stageDefinition} />}
    </div>
  );
};

export default Extension;
