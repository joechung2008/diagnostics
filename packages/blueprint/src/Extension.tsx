import Configuration from "./Configuration";
import StageDefinition from "./StageDefinition";

const Extension: React.FC<ExtensionProps> = ({
  config,
  extensionName,
  stageDefinition,
}) => {
  return (
    <div id="extension" className="extension-root grow">
      <h2 className="bp6-heading">{extensionName}</h2>
      {config && <Configuration config={config} />}
      {stageDefinition && <StageDefinition stageDefinition={stageDefinition} />}
    </div>
  );
};

export default Extension;
