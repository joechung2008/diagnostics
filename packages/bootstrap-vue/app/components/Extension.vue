<template>
  <div class="extension">
    <h1>{{ extensionName }}</h1>
    <Configuration v-if="config" :config="config" />
    <StageDefinition v-if="stageDefinition" :stage-definition="stageDefinition" />
  </div>
</template>

<script setup lang="ts">
import type { ExtensionProps } from "@/types/Extension";

defineProps<ExtensionProps>();
</script>

<style scoped>
.extension {
  display: flex;
  flex: 1;
  flex-direction: column;
  gap: 0.5rem;
  max-height: max-content;
  overflow-y: auto;
  padding: 0.25rem;
}
</style>
