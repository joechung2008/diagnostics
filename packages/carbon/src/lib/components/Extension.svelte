<script lang="ts">
	import Configuration from './Configuration.svelte';
	import StageDefinition from './StageDefinition.svelte';

	let { extensionName, config, stageDefinition }: ExtensionProps = $props();
</script>

<div class="extension-root" role="tabpanel" tabindex="0">
	<h1>{extensionName}</h1>
	{#if config}
		<Configuration {config} />
	{/if}
	{#if stageDefinition}
		<StageDefinition {stageDefinition} />
	{/if}
</div>

<style>
	.extension-root {
		display: flex;
		flex: 1;
		flex-direction: column;
		gap: 0.5rem;
		overflow-y: auto;
		padding: 0.5rem;
	}
</style>
