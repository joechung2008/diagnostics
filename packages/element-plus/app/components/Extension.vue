<template>
    <div class="extension">
        <h1>{{ extensionName }}</h1>
        <Configuration v-if="config" :config="config" />
        <StageDefinition v-if="stageDefinition" :stage-definition="stageDefinition" />
    </div>
</template>

<script setup lang="ts">
import type { ExtensionProps } from "@/types"

defineProps<ExtensionProps>()
</script>

<style scoped>
.extension {
    display: flex;
    flex: 1;
    flex-direction: column;
    gap: 0.5rem;
    max-height: max-content;
    overflow-y: auto;
    padding: 0.25rem;
}

h1 {
    margin: 0;
}
</style>
