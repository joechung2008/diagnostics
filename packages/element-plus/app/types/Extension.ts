import type { ExtensionInfo } from "./ExtensionInfo"
import type { ExtensionError } from "./ExtensionError"

export type Extension = ExtensionInfo | ExtensionError
