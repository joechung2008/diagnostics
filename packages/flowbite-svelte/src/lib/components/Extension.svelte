<script lang="ts">
  import { Heading } from "flowbite-svelte";
  import Configuration from "./Configuration.svelte";
  import StageDefinition from "./StageDefinition.svelte";

  let { config, extensionName, stageDefinition }: ExtensionProps = $props();
</script>

<div class="flex flex-grow flex-col gap-0.5 overflow-y-auto p-1">
  <Heading tag="h1">
    {extensionName}
  </Heading>
  {#if config}
    <Configuration {config} />
  {/if}
  {#if stageDefinition}
    <StageDefinition {stageDefinition} />
  {/if}
</div>
