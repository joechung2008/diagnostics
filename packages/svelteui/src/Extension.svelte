<script lang="ts">
  import Configuration from './Configuration.svelte'
  import StageDefinition from './StageDefinition.svelte'

  export let extensionName: string
  export let config: Record<string, string> | undefined = undefined
  export let stageDefinition: Record<string, string[]> | undefined = undefined
</script>

<div class="extension" role="tabpanel" tabindex={0}>
  <h1 class="title">{extensionName}</h1>
  <div class="content">
    {#if config}
      <Configuration {config} />
    {/if}
    {#if stageDefinition}
      <StageDefinition {stageDefinition} />
    {/if}
  </div>
</div>

<style scoped>
  .extension {
    display: flex;
    flex: 1;
    flex-direction: column;
    overflow-y: auto;
  }

  .title {
    font-size: 1.5rem;
    font-weight: 600;
    margin-bottom: 1rem;
  }

  .content {
    display: flex;
    flex-direction: column;
    gap: 1.5rem;
  }
</style>
