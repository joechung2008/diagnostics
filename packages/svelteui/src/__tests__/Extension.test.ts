import { describe, it, expect } from 'vitest'
import { render } from '@testing-library/svelte'
import Extension from '../Extension.svelte'

describe('Extension', () => {
  it('should render extension with only name', () => {
    const { container } = render(Extension, {
      props: {
        extensionName: 'TestExtension'
      }
    })

    expect(container).toMatchSnapshot()
  })

  it('should render extension with name and config', () => {
    const { container } = render(Extension, {
      props: {
        extensionName: 'DatabaseExtension',
        config: {
          'db.host': 'localhost',
          'db.port': '5432',
          'db.name': 'myapp'
        }
      }
    })

    expect(container).toMatchSnapshot()
  })

  it('should render extension with name and stage definition', () => {
    const { container } = render(Extension, {
      props: {
        extensionName: 'ApiExtension',
        stageDefinition: {
          development: ['localhost:3000', 'dev.api.com'],
          production: ['api.com', 'api.example.com']
        }
      }
    })

    expect(container).toMatchSnapshot()
  })

  it('should render extension with name, config, and stage definition', () => {
    const { container } = render(Extension, {
      props: {
        extensionName: 'FullFeaturedExtension',
        config: {
          'api.version': 'v1',
          'auth.enabled': 'true',
          'rate.limit': '1000'
        },
        stageDefinition: {
          development: ['localhost:8080'],
          staging: ['staging-api.com'],
          production: ['api.production.com', 'api-backup.production.com']
        }
      }
    })

    expect(container).toMatchSnapshot()
  })

  it('should render extension with empty config object', () => {
    const { container } = render(Extension, {
      props: {
        extensionName: 'EmptyConfigExtension',
        config: {}
      }
    })

    expect(container).toMatchSnapshot()
  })

  it('should render extension with empty stage definition object', () => {
    const { container } = render(Extension, {
      props: {
        extensionName: 'EmptyStageExtension',
        stageDefinition: {}
      }
    })

    expect(container).toMatchSnapshot()
  })

  it('should render extension with special characters in name', () => {
    const { container } = render(Extension, {
      props: {
        extensionName: 'Test_Extension-v2.1.0'
      }
    })

    expect(container).toMatchSnapshot()
  })
})
