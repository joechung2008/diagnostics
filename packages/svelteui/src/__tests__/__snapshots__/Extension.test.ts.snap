// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Extension > should render extension with empty config object 1`] = `
<div>
  <div
    class="extension svelte-19guq2i"
    role="tabpanel"
    tabindex="0"
  >
    <h1
      class="title svelte-19guq2i"
    >
      EmptyConfigExtension
    </h1>
     
    <div
      class="content svelte-19guq2i"
    >
      <div>
        <h2
          class="svelte-1riupzq"
          id="configuration"
        >
          Configuration
        </h2>
         
        <div
          class="table-container svelte-1riupzq"
        >
          <table
            aria-labelledby="configuration"
            class="svelte-1riupzq"
          >
            <thead>
              <tr>
                <th
                  class="svelte-1riupzq"
                >
                  Key
                </th>
                 
                <th
                  class="svelte-1riupzq"
                >
                  Value
                </th>
              </tr>
            </thead>
             
            <tbody />
          </table>
        </div>
      </div>
       
    </div>
  </div>
</div>
`;

exports[`Extension > should render extension with empty stage definition object 1`] = `
<div>
  <div
    class="extension svelte-19guq2i"
    role="tabpanel"
    tabindex="0"
  >
    <h1
      class="title svelte-19guq2i"
    >
      EmptyStageExtension
    </h1>
     
    <div
      class="content svelte-19guq2i"
    >
       
      <div>
        <h2
          class="svelte-1riupzq"
          id="stage-definition"
        >
          Stage Definitions
        </h2>
         
        <div
          class="table-container svelte-1riupzq"
        >
          <table
            aria-labelledby="stage-definition"
            class="svelte-1riupzq"
          >
            <thead>
              <tr>
                <th
                  class="svelte-1riupzq"
                >
                  Key
                </th>
                 
                <th
                  class="svelte-1riupzq"
                >
                  Value
                </th>
              </tr>
            </thead>
             
            <tbody />
          </table>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Extension > should render extension with name and config 1`] = `
<div>
  <div
    class="extension svelte-19guq2i"
    role="tabpanel"
    tabindex="0"
  >
    <h1
      class="title svelte-19guq2i"
    >
      DatabaseExtension
    </h1>
     
    <div
      class="content svelte-19guq2i"
    >
      <div>
        <h2
          class="svelte-1riupzq"
          id="configuration"
        >
          Configuration
        </h2>
         
        <div
          class="table-container svelte-1riupzq"
        >
          <table
            aria-labelledby="configuration"
            class="svelte-1riupzq"
          >
            <thead>
              <tr>
                <th
                  class="svelte-1riupzq"
                >
                  Key
                </th>
                 
                <th
                  class="svelte-1riupzq"
                >
                  Value
                </th>
              </tr>
            </thead>
             
            <tbody>
              <tr>
                <td
                  class="svelte-1riupzq"
                >
                  db.host
                </td>
                 
                <td
                  class="svelte-1riupzq"
                >
                  localhost
                </td>
                 
              </tr>
              <tr>
                <td
                  class="svelte-1riupzq"
                >
                  db.port
                </td>
                 
                <td
                  class="svelte-1riupzq"
                >
                  5432
                </td>
                 
              </tr>
              <tr>
                <td
                  class="svelte-1riupzq"
                >
                  db.name
                </td>
                 
                <td
                  class="svelte-1riupzq"
                >
                  myapp
                </td>
                 
              </tr>
            </tbody>
          </table>
        </div>
      </div>
       
    </div>
  </div>
</div>
`;

exports[`Extension > should render extension with name and stage definition 1`] = `
<div>
  <div
    class="extension svelte-19guq2i"
    role="tabpanel"
    tabindex="0"
  >
    <h1
      class="title svelte-19guq2i"
    >
      ApiExtension
    </h1>
     
    <div
      class="content svelte-19guq2i"
    >
       
      <div>
        <h2
          class="svelte-1riupzq"
          id="stage-definition"
        >
          Stage Definitions
        </h2>
         
        <div
          class="table-container svelte-1riupzq"
        >
          <table
            aria-labelledby="stage-definition"
            class="svelte-1riupzq"
          >
            <thead>
              <tr>
                <th
                  class="svelte-1riupzq"
                >
                  Key
                </th>
                 
                <th
                  class="svelte-1riupzq"
                >
                  Value
                </th>
              </tr>
            </thead>
             
            <tbody>
              <tr>
                <td
                  class="svelte-1riupzq"
                >
                  development
                </td>
                 
                <td
                  class="svelte-1riupzq"
                >
                  localhost:3000, dev.api.com
                </td>
                 
              </tr>
              <tr>
                <td
                  class="svelte-1riupzq"
                >
                  production
                </td>
                 
                <td
                  class="svelte-1riupzq"
                >
                  api.com, api.example.com
                </td>
                 
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Extension > should render extension with name, config, and stage definition 1`] = `
<div>
  <div
    class="extension svelte-19guq2i"
    role="tabpanel"
    tabindex="0"
  >
    <h1
      class="title svelte-19guq2i"
    >
      FullFeaturedExtension
    </h1>
     
    <div
      class="content svelte-19guq2i"
    >
      <div>
        <h2
          class="svelte-1riupzq"
          id="configuration"
        >
          Configuration
        </h2>
         
        <div
          class="table-container svelte-1riupzq"
        >
          <table
            aria-labelledby="configuration"
            class="svelte-1riupzq"
          >
            <thead>
              <tr>
                <th
                  class="svelte-1riupzq"
                >
                  Key
                </th>
                 
                <th
                  class="svelte-1riupzq"
                >
                  Value
                </th>
              </tr>
            </thead>
             
            <tbody>
              <tr>
                <td
                  class="svelte-1riupzq"
                >
                  api.version
                </td>
                 
                <td
                  class="svelte-1riupzq"
                >
                  v1
                </td>
                 
              </tr>
              <tr>
                <td
                  class="svelte-1riupzq"
                >
                  auth.enabled
                </td>
                 
                <td
                  class="svelte-1riupzq"
                >
                  true
                </td>
                 
              </tr>
              <tr>
                <td
                  class="svelte-1riupzq"
                >
                  rate.limit
                </td>
                 
                <td
                  class="svelte-1riupzq"
                >
                  1000
                </td>
                 
              </tr>
            </tbody>
          </table>
        </div>
      </div>
       
      <div>
        <h2
          class="svelte-1riupzq"
          id="stage-definition"
        >
          Stage Definitions
        </h2>
         
        <div
          class="table-container svelte-1riupzq"
        >
          <table
            aria-labelledby="stage-definition"
            class="svelte-1riupzq"
          >
            <thead>
              <tr>
                <th
                  class="svelte-1riupzq"
                >
                  Key
                </th>
                 
                <th
                  class="svelte-1riupzq"
                >
                  Value
                </th>
              </tr>
            </thead>
             
            <tbody>
              <tr>
                <td
                  class="svelte-1riupzq"
                >
                  development
                </td>
                 
                <td
                  class="svelte-1riupzq"
                >
                  localhost:8080
                </td>
                 
              </tr>
              <tr>
                <td
                  class="svelte-1riupzq"
                >
                  staging
                </td>
                 
                <td
                  class="svelte-1riupzq"
                >
                  staging-api.com
                </td>
                 
              </tr>
              <tr>
                <td
                  class="svelte-1riupzq"
                >
                  production
                </td>
                 
                <td
                  class="svelte-1riupzq"
                >
                  api.production.com, api-backup.production.com
                </td>
                 
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Extension > should render extension with only name 1`] = `
<div>
  <div
    class="extension svelte-19guq2i"
    role="tabpanel"
    tabindex="0"
  >
    <h1
      class="title svelte-19guq2i"
    >
      TestExtension
    </h1>
     
    <div
      class="content svelte-19guq2i"
    >
       
    </div>
  </div>
</div>
`;

exports[`Extension > should render extension with special characters in name 1`] = `
<div>
  <div
    class="extension svelte-19guq2i"
    role="tabpanel"
    tabindex="0"
  >
    <h1
      class="title svelte-19guq2i"
    >
      Test_Extension-v2.1.0
    </h1>
     
    <div
      class="content svelte-19guq2i"
    >
       
    </div>
  </div>
</div>
`;
